### DISCUSSION TITLE
About DA challenge for generic commitment type

### DISCUSSION THREAD

**Discussion by zhiqiangxu** [ethereum-optimism/developers#576] [General]
I'm interested in understanding more about the current state of DA challenges. 

From what I've gathered, it appears that DA challenges are currently feasible for keccak256 commitments, but it seems they're impossible for generic commitments. 

Is this understanding is correct? If it's true, does it mean that most alt-DAs like celestia will not be DA challenge-able?

**Comment by tynes** [comment]
This is correct @zhiqiangxu. Celestia and EiganDA will have their own security mechanisms

**Comment by zhiqiangxu** [comment]
@tynes So using Celestia and EiganDA for plasma implies some trusting assumptions about the DA, and since DA challenge is not designed for them, in the worst case the data withholding issue may happen without any penalties, is this correct?

**Comment by upnodedev** [comment]
We are building an Alt-DA system that enables third-party DAs like Celestia and EigenDA to utilize the Keccak256 commitment. This allows us to leverage the security of Keccak256 commitment challenge logic even when using these alternative DA solutions.

https://github.com/upnodedev/op-alt-da

**Comment by zhiqiangxu** [comment]
@upnodedev That's cool!

Out of curiosity, is this how it works?

1. eigen_tx_hash = send_tx(preimage)
2. send `keccack(preimage) ++ eigen_tx_hash` to L1
3. If can't fetch the `preimage` with `eigen_tx_hash`, then challenge `keccack(preimage)`.
